go/logger/buffer.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package logger
+
+import (
+	"bufio"
+	"io"
+	"sync"
+	"time"
+)
+
+type triggerableTimer struct {
+	C          chan struct{}
+	timer      *time.Timer
+	sentinelCh chan struct{}
+	shutdownCh chan struct{}
+}
+
+func newTriggerableTimer(d time.Duration) *triggerableTimer {
```

Comment:
I think we should trigger this with no duration, otherwise we'll block in the `select` below, on `<-t.sentinelCh`, until a `ResetIfStopped` comes in. Then it'll fire right away.